### PR TITLE
python310Packages.bayespy: Fix tests

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Fixing tests for Python module `bayespy`  and cleanup code

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/bayespy/default.nix [hunk 1]
```diff
@@ -15,12 +15,25 @@ buildPythonPackage rec {
     sha256 = "ed0057dc22bd392df4b3bba23536117e1b2866e3201b12c5a37428d23421a5ba";
   };
 
-  checkInputs = [ pytest nose glibcLocales ];
+  patches = [
+    # Change from scipy to locally defined epsilon
+    # https://github.com/bayespy/bayespy/pull/126
+    (fetchpatch {
+      url = "https://patch-diff.githubusercontent.com/raw/bayespy/bayespy/pull/126.patch";
```
Comment (risicle) [inline pkgs/development/python-modules/bayespy/default.nix:22]: In the case of a PR that's been merged it's usually best to use the merge-commit from the PR. In this case https://github.com/bayespy/bayespy/commit/9be53bada763e19c2b6086731a6aa542ad33aad0.patch would be the url to use. In theory PRs can be mutated whereas commits should be immutable.

It's also nice to give patches names, in this case I would use `name = "locally-defined-epsilon.patch";`.

**Comment by onny** [comment]
@risicle Fixed it, thank you :)

**Comment by risicle** [comment]
This also seems to build for me on macos 10.15 & nixos aarch64, so would be good to get rid of those `broken` labels too.

**Comment by onny** [comment]
@risicle Sounds good, removed the broken tags.

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15, nixos x86_64 & aarch64.

**Pull request merged**